lib/src/main/java/uk/gov/di/ipv/cri/passport/library/persistence/item/converter/JacksonAttributeConverter.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.di.ipv.cri.passport.library.persistence.item.converter;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import software.amazon.awssdk.enhanced.dynamodb.AttributeConverter;
+import software.amazon.awssdk.enhanced.dynamodb.AttributeValueType;
+import software.amazon.awssdk.enhanced.dynamodb.EnhancedType;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import uk.gov.di.ipv.cri.passport.library.annotations.ExcludeFromGeneratedCoverageReport;
+
+import java.io.UncheckedIOException;
+
+@ExcludeFromGeneratedCoverageReport
+public class JacksonAttributeConverter <T> implements AttributeConverter<T> {
+
+    private final Class<T> clazz;
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    static {
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, true);
+    }
+
+    public JacksonAttributeConverter(Class<T> clazz) {
+        this.clazz = clazz;
+    }
+
+    @Override
+    public AttributeValue transformFrom(T input) {
+        try {
+            return AttributeValue
+                    .builder()
+                    .s(objectMapper.writeValueAsString(input))
```

Comment:
I've run this against dev account and it seems to work, a `PassportCheckDao` object is serailized and mapped to Dynamodb as follows:
```
        {
            "attributes": {
                "S": "{\"passportNumber\":\"passport-number\",\"surname\":\"surname\",\"forenames\":[\"family-name\"],\"dateOfBirth\":\"1900-01-01\",\"expiryDate\":\"2025-02-02\",\"correlationId\":\"f73339da-e147-4f75-b21c-34558f51bb1d\
",\"requestId\":\"aed0151a-27fa-45df-b16e-007a8badfe6c\",\"timestamp\":\"2022-02-08T08:41:53.956+0000\",\"dcsResponse\":{\"correlationId\":\"6c4a3b26-c8b6-460d-9d31-8e3ea0c87931\",\"requestId\":\"8a74566c-6c6d-4b52-a0ea-054c53dea34c\",\
"error\":false,\"valid\":true,\"errorMessage\":null}}"
            },
            "gpg45Score": {
                "S": "{\"evidence\":{\"strength\":5,\"validity\":5}}"
            },
            "resourceId": {
                "S": "resource-id-2"
            }
        },
```
So its correctly sets the `resourceId` partition key since it seems to take each top level key in the JSON string and set them as String attributes. I'm assuming that not parsing the values into AttributeValues will make it harder to query this data in DynamoDb by anything other than the partition key, for example by name or passport number where we were previously mapping as something like
```
        {
            "resourceId": {
                "S": "1c8e0b66-656c-4380-bbdd-c76a1e1948cc"
            },
            "attributes": {
                "M": {
                    "passportNumber": {
                        "S": "123456789"
                    },
                    "expiryDate": {
                        "S": "2024-01-01"
                    },
                    "dateOfBirth": {
                        "S": "1900-01-01"
                    },
                    "forenames": {
                        "SS": [
                            "first-name"
                        ]
                    },
                    "surname": {
                        "S": "surname"
                    }
                }
            }
        },
```

The benefit for the time being is that this is very flexible to changes in our data model which is in flux so maybe we use this until things settle down and if we need to map the full tree to AttributesValues later on we do so. I should caveat that I'm only on module 2 of a DynamoDb deep dive course and haven't reached the "best practices" section yet which might have some words of wisdom on this approach...)

To get this to work I had to add no arg constructors to a few classes so Jackson could serialise them, and add the `JavaTimeModule` to the object mapper above
```
    ..... new ObjectMapper().registerModule(new JavaTimeModule());
```